### ISSUE TITLE
Hash instead password in config clickhouse-client.xml

### ISSUE THREAD

**Issue by lavasser** [ClickHouse/ClickHouse#36694]
Hello!
It would be nice to add hash instead password in config clickhouse-client.xml
It will be more secure.

For example:
```xml
<config>
<host>127.0.0.1</host>
    <port>9440</port>
    <secure>1</secure>
    <user>dbuser</user>
    <password_sha256_hex>080fe732956685df3c63a4483004dd7cf512e8502570d4264712b9d3afda1422</password_sha256_hex>
</config>
```

**Comment by rkozlo** [comment]
I don't think it is good idea. Hash is often visible for others and case when someone would get this hash and simply use it to logging in seems to be security vulnerability

**Comment by den-crane** [comment]
A client needs a password to connect to a server.
It's impossible to restore a password from a hash.
All idea about hashing password is to make impossible to restore a password from a hash.

**Comment by kocmuk58** [comment]
Using a hash instead of a password is a good idea.
For example, this is implemented in Opensearch (Elasticsearch)
https://opensearch.org/docs/latest/security-plugin/configuration/yaml/

**Comment by den-crane** [comment]
> Using a hash instead of a password is a good idea. For example, this is implemented in Opensearch (Elasticsearch) https://opensearch.org/docs/latest/security-plugin/configuration/yaml/

OK, Genius:
Here is the site https://github.com/ClickHouse/ClickHouse/issues/36694#issuecomment-1111027708
User: `demo` 
hash for demo user password is `eb9c26baee47f19e4993a77bca936d0ff09e355a82d3db79bf154ebff1a80604`

Try to connect and execute select

**Comment by den-crane** [comment]
This configuration file is for a CLIENT. 
THE CLIENT uses this configuration file to connect to a SERVER.
SERVER configuration files can store CLEAR passwords and HASHES password_sha256_hex 
https://clickhouse.com/docs/en/operations/settings/settings-users#user-namepassword
